tests/ghost_test.go

```diff
@@ -0,0 +1,196 @@
+package tests
+
+import (
+	"database/sql"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/bytebase/bytebase/resources/mysql"
+	"github.com/github/gh-ost/go/base"
+	"github.com/github/gh-ost/go/logic"
+	ghostsql "github.com/github/gh-ost/go/sql"
+	_ "github.com/go-sql-driver/mysql"
+)
+
+type config struct {
+	host           string
+	port           int
+	user           string
+	password       string
+	database       string
+	table          string
+	alterStatement string
+	execute        bool
+}
+
+const localhost = "127.0.0.1"
+
+const (
+	allowedRunningOnMaster              = true
+	concurrentCountTableRows            = true
+	hooksStatusIntervalSec              = 60
+	replicaServerID                     = 99999
+	heartbeatIntervalMilliseconds       = 100
+	niceRatio                           = 0
+	chunkSize                           = 1000
+	dmlBatchSize                        = 10
+	maxLagMillisecondsThrottleThreshold = 1500
+	defaultNumRetries                   = 60
+	cutoverLockTimoutSeconds            = 3
+	exponentialBackoffMaxInterval       = 64
+)
+
+func newMigrationContext(config config) (*base.MigrationContext, error) {
+	migrationContext := base.NewMigrationContext()
+	migrationContext.InspectorConnectionConfig.Key.Hostname = config.host
+	migrationContext.InspectorConnectionConfig.Key.Port = config.port
+	migrationContext.CliUser = config.user
+	migrationContext.CliPassword = config.password
+	migrationContext.DatabaseName = config.database
+	migrationContext.OriginalTableName = config.table
+	migrationContext.AlterStatement = config.alterStatement
+	migrationContext.Noop = !config.execute
```

Comment:
Does Noop mean dryRun? if so, then we should name execute to dryRun

Reply:
I'll rename it to noop to be consistent with migrationContext.